Pr #3

jamezmca opened the pull request:
Using Cloudinary to host images where the fetched responses from Cloudinary are passed into postMessage from the Input.js file.  After being passed into postMessage, the images are passed through the Messages.js file into the SenderBubble.js file. The styling is dependent on the number of images sent and whether or not text is included in the message :P

<img width="449" alt="Screen Shot 2021-09-11 at 5 46 23 PM" src="https://user-images.githubusercontent.com/77163783/132967179-283ebf59-b18d-47a9-8bd8-a42e633a5d13.png">
<img width="353" alt="Screen Shot 2021-09-11 at 5 48 43 PM" src="https://user-images.githubusercontent.com/77163783/132967182-15892bb2-1203-4f02-bb6f-5c9f8cecd0b9.png">



![Screen Shot 2021-09-11 at 5 50 28 PM](https://user-images.githubusercontent.com/77163783/132967235-70b6c4e4-d351-4f70-894b-1c1591a5c0ee.png)

jamezmca committed:
finished the file upload button and inputs

using axios to post data to cloudinary following uploading the images.

jamezmca committed:
Cloudinary functional and sending images for one image upload

need to implement double image upload and size and style images appropriately

jamezmca committed:
Multiple images are now uploaded

just want to display number of images sent and size the images appropriately

jamezmca committed:
uploading two images now functional

just need to finish styling

jamezmca committed:
styling the different image layouts

jamezmca committed:
Finished styling the sent images :)

jamezmca committed:
Update Input.js

remove unnecessary styles

jamezmca committed:
Update SenderBubble.js

just made a minor update to the alt prop on an img tag

sonipranjal reviewed (commented):
Good job!

- Always remember to clean up the code before making a pull request.
- Now, we can select and upload one image at a time, but as described in mocks, we should be able to select more than one image.

I have added some comments below in more detail; please address them and let us know on the slack channel when this PR is ready for another review!

jamezmca committed:
Updated input.js so that the input can accept multiple files

used promise.all as there are multiple await statements

also removed console.logs

moved api keys to a .env file

etchao reviewed (approved):
There are still a couple areas that could be improved and the UI for sending images is not consistent with the spec (the spec shows the text below the image if both are present), but otherwise this PR can be merged 🙂

jamezmca commented on client/src/components/ActiveChat/Input.js:
```diff
@@ -1,26 +1,69 @@
   root: {
     justifySelf: "flex-end",
-    marginTop: 15
+    marginTop: 15,
```
Thanks for that comment - I was unaware but I will be sure to implement this in future. Also regarding the text with image sending, I did implement the text displaying beneath the image for the case where only a single image is sent. If two images are sent, it displays the text above. Does it work the same for you?

Thanks again for your feedback :)

etchao replied:
Ah yes - my mistake about the UI, it looks right to me. I made the mistake of testing with a white image and saw it incorrectly. Sorry about that!

jamezmca replied:
No worries at all :) thanks for your feedback!

jamezmca committed:
Update Input.js

A quick patchfix
